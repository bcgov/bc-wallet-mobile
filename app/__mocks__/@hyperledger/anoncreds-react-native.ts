const AnonCredsReactNative = {
  install: jest.fn(),
}

export default AnonCredsReactNative
