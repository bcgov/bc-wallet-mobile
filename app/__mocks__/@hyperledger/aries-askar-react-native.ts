const AriesAskarReactNative = {
  install: jest.fn(),
}

export default AriesAskarReactNative
