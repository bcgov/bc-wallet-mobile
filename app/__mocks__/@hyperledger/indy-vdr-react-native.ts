const IndyVDRReactNative = {
  install: jest.fn(),
}

export default IndyVDRReactNative
