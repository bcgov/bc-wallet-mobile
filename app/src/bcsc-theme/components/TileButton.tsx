import { testIdWithKey, useTheme } from '@bifold/core'
import { Text, View, Image, StyleSheet, ImageSourcePropType, ViewStyle, TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'

export type TileButtonProps = {
  onPress: () => void
  testIDKey: string
  accessibilityLabel: string
  actionText: string
  description: string
  imgSrc?: ImageSourcePropType
  style?: ViewStyle
}

const TileButton: React.FC<TileButtonProps> = ({
  onPress,
  testIDKey,
  accessibilityLabel,
  actionText,
  description,
  imgSrc,
  style,
}: TileButtonProps) => {
  const { ColorPallet, TextTheme, Spacing } = useTheme()
  const styles = StyleSheet.create({
    tile: {
      borderRadius: Spacing.sm,
      borderWidth: 1,
      borderColor: ColorPallet.notification.infoBorder,
      backgroundColor: ColorPallet.notification.info,
      padding: 24,
    },
    actionText: {
      ...TextTheme.bold,
      color: ColorPallet.brand.primary,
    },
    description: {
      ...TextTheme.normal,
      marginTop: Spacing.md,
    },
  })

  return (
    <TouchableOpacity
      style={style}
      onPress={onPress}
      testID={testIdWithKey(testIDKey)}
      accessibilityLabel={accessibilityLabel}
    >
      <View style={styles.tile}>
        <Text style={styles.actionText}>
          {actionText}
          <Icon size={20} color={ColorPallet.brand.primary} name={'arrow-right-thin'} />
        </Text>
        <Text style={styles.description}>{description}</Text>
        {imgSrc ? (
          <Image source={imgSrc} style={{ flex: 1, height: 100, width: 150, marginTop: 16 }} resizeMode={'contain'} />
        ) : null}
      </View>
    </TouchableOpacity>
  )
}

export default TileButton
