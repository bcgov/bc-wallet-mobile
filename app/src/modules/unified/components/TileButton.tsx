import { testIdWithKey, useTheme } from '@hyperledger/aries-bifold-core'
import { Text, View, Image, StyleSheet, Pressable, ImageSourcePropType, ViewStyle } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'

export type TileButtonProps = {
  onPress: () => void
  testIDKey: string
  accessibilityLabel: string
  actionText: string
  description: string
  imgSrc?: ImageSourcePropType
  style?: ViewStyle
}

const TileButton: React.FC<TileButtonProps> = ({
  onPress,
  testIDKey,
  accessibilityLabel,
  actionText,
  description,
  imgSrc,
  style,
}: TileButtonProps) => {
  const { ColorPallet, TextTheme } = useTheme()
  const styles = StyleSheet.create({
    tile: {
      borderRadius: 4,
      backgroundColor: ColorPallet.notification.info,
      padding: 24,
      shadowOffset: { width: 0, height: 4 },
      shadowRadius: 4,
      shadowColor: ColorPallet.grayscale.lightGrey,
      shadowOpacity: 0.9,
      elevation: 2, // for older android versions
    },
    actionText: {
      ...TextTheme.bold,
      color: ColorPallet.brand.primary,
    },
    description: {
      ...TextTheme.normal,
      marginTop: 16,
    },
  })

  return (
    <Pressable
      style={style}
      onPress={onPress}
      testID={testIdWithKey(testIDKey)}
      accessibilityLabel={accessibilityLabel}
    >
      <View style={styles.tile}>
        <Text style={styles.actionText}>
          {actionText}
          <Icon size={20} color={ColorPallet.brand.primary} name={'arrow-right-thin'} />
        </Text>
        <Text style={styles.description}>{description}</Text>
        {imgSrc ? (
          <Image source={imgSrc} style={{ flex: 1, height: 100, width: 150, marginTop: 16 }} resizeMode={'contain'} />
        ) : null}
      </View>
    </Pressable>
  )
}

export default TileButton
