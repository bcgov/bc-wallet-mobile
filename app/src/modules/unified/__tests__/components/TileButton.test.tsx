import { testIdWithKey } from '@hyperledger/aries-bifold-core'
import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import TileButton from '../../components/TileButton'

describe('TileButton Component', () => {
  const onPress = jest.fn()
  const actionText = 'Action text'
  const description =
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quasi et consectetur iusto error aliquam'
  const testIDKey = 'TileButton'
  const accessibilityLabel = 'Card'

  beforeEach(() => {
    jest.resetAllMocks()
  })

  test('renders correctly', () => {
    const tree = render(
      <TileButton
        onPress={onPress}
        actionText={actionText}
        description={description}
        testIDKey={testIDKey}
        accessibilityLabel={accessibilityLabel}
      />
    )
    expect(tree).toMatchSnapshot()
  })

  test('test ID is present and press handler works', () => {
    const { getByTestId } = render(
      <TileButton
        onPress={onPress}
        actionText={actionText}
        description={description}
        testIDKey={testIDKey}
        accessibilityLabel={accessibilityLabel}
      />
    )

    const tileButton = getByTestId(testIdWithKey(testIDKey))
    expect(tileButton).toBeDefined()
    fireEvent(tileButton, 'press')
    expect(onPress).toHaveBeenCalledTimes(1)
  })
})
