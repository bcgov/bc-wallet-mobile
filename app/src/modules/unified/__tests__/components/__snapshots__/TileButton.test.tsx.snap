// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileButton Component renders correctly 1`] = `
<View
  accessibilityLabel="Card"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="com.ariesbifold:id/TileButton"
>
  <View
    style={
      {
        "backgroundColor": "#313132",
        "borderRadius": 4,
        "elevation": 2,
        "padding": 24,
        "shadowColor": "#D3D3D3",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.9,
        "shadowRadius": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#42803E",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Action text
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#42803E",
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ó±¦°
      </Text>
    </Text>
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "normal",
          "marginTop": 16,
        }
      }
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quasi et consectetur iusto error aliquam
    </Text>
  </View>
</View>
`;
